### PR TITLE
Config update for msi

### PR DESCRIPTION
**Pull request by JanPrusinowski**

### CODE REVIEW

File: lib/secure-boot-lib.robot
Comment (krystian-hebel) [inline lib/secure-boot-lib.robot:151]: Nice catch, but please add a comment why this is needed, otherwise someone may try to remove it in the future.

File: platform-configs/include/msi-z690-common.robot
Comment (krystian-hebel) [inline platform-configs/include/msi-z690-common.robot:134]: Is this the correct order? I'd expect it to be before `Rte Power On`. You first have to plug the power cord, and then press the button.

File: platform-configs/msi-pro-z690-a-wifi-ddr4.robot
Comment (krystian-hebel) [inline platform-configs/msi-pro-z690-a-wifi-ddr4.robot:31]: We have two such boards in the lab, but these are valid for only one of them. For now, please add a comment mentioning this fact and CPU model for which those values are valid, we will have to somehow handle it in the future. The other platform isn't fully remotely available anyway.

File: lib/options/uefi-setup-menu.robot
Comment (krystian-hebel) [reply lib/options/uefi-setup-menu.robot:133]: In fact, `Measure * Time` keywords also don't fit here :thinking:

File: dasharo-compatibility/cpu-cores-count.robot
Comment (krystian-hebel) [inline dasharo-compatibility/cpu-cores-count.robot:61]: (applies to all changes in this file)

Why is this needed? `Set UEFI Option` ends with `Save Changes And Reset`, is it broken?

**Review by krystian-hebel — CHANGES_REQUESTED** [review]

File: keywords.robot
Comment (macpijan) [reply keywords.robot:1732]: @krystian-hebel For the devices with PiKVM we simply look for the PiKVM device in the boot menu to evaluate whether the boot device is detected. For others, we have to use other ways, which may or may not work presently.

File: lib/bios/menus.robot
Comment (macpijan) [inline lib/bios/menus.robot:416]: I do not get it. If serial is not enabled, how this code can work, as it reads from serial? This seems pointless, we simply have to modify the test binary via DCU prior running tests, which is already documented.

Unless I'm missing something here, then please correct me.

Comment (macpijan) [reply lib/bios/menus.robot:416]: Ok, I think I do get it on the second review. We want to enable serial after we reset settings to default, to ensure the serial remains enabled, even if it has just been disabled by resetting.

Comment (krystian-hebel) [reply lib/bios/menus.robot:416]: This is part of `Reset To Defaults`, and default is disabled. DCU doesn't change the default, only the current setting.

Comment (macpijan) [reply lib/bios/menus.robot:416]: If this is the case, it is not necessarily tied to PiKVM directly in any way.

This is simply needed for devices where serial is disabled by default. Which may or may not be related to the fact that they do have PiKVM connected as well.

What is more, it should not hurt to make sure this options is selected even on devices where serial is enabled after resetting to defaults.

File: lib/bios/menus.robot
Comment (macpijan) [inline lib/bios/menus.robot:446]: I checked and it does not work for me, it sets highlight to the `Reset` instead

File: lib/bios/menus.robot
Comment (macpijan) [inline lib/bios/menus.robot:434]: I tried and it needed 2 iterations to go back to the main menu with 1 nested screen, instead of one.

File: lib/bios/menus.robot
Comment (macpijan) [inline lib/bios/menus.robot:432]: You are aiming very high choosing to develop such a keyword.

It will be very tricky to get it right, depending on how it's been called.

In the first iteration this keyword relies on whatever was left in the telnet buffer by the previous keyword. The fact that you kewyord runs at all is simply because you chose to read until `save screenshot`, while in other menu handling keywords we read until `Esc=Exit` (`Save screenshot` has been added later at some point).

File: lib/bios/menus.robot
Comment (macpijan) [reply lib/bios/menus.robot:446]: The problem is, that after 1st ESC what you get over telnet when reading until `save screenshot` is the menu, but with no help text visible:

```
QEMU x86 q35/ich9
 QEMU Virtual CPU version 2.5+                       0.00 GHz
 Dasharo (coreboot+UEFI) v0.2.0                      2048 MB RAM @ 200 MHz



   Select Language            <Standard English>

 > User Password Management
 > Device Manager
 > Dasharo System Features
 > One Time Boot
 > Boot Maintenance Manager

   Continue
   Reset





  LCtrl+LAlt+F12=Save screenshot
  ```

File: platform-configs/include/msi-z690-common.robot
Comment (macpijan) [inline platform-configs/include/msi-z690-common.robot:93]: That was disabled mostly because of the problems with serial being disabled after resetting. We can rather quickly test these manually after we finish automated tests.

File: lib/bios/menus.robot
Comment (macpijan) [reply lib/bios/menus.robot:446]: I suggest that we move it to a separate task, as getting the `Reset to defaults` to work on MSI is going to be tricky...

I suggest we stick with the original automation plan we've had in the configs, and focus on getting these to work, not adding more.

File: platform-configs/include/msi-z690-common.robot
Comment (macpijan) [reply platform-configs/include/msi-z690-common.robot:93]: see: https://github.com/Dasharo/open-source-firmware-validation/pull/530/files#r1806136564

File: platform-configs/include/msi-z690-common.robot
Comment (macpijan) [inline platform-configs/include/msi-z690-common.robot:43]: If it relies on `rest to defaults`, we may give up on running these tests automatically right now as well: https://github.com/Dasharo/open-source-firmware-validation/pull/530/files#r1806136564

Or at least leave it until we have all of the "old" automation working in place.

File: dasharo-compatibility/esp-scanning.robot
Comment (macpijan) [inline dasharo-compatibility/esp-scanning.robot:30]: `Power On` is already in the `Clear Out EFI Partition` - seems redundant.

File: platform-configs/include/msi-z690-common.robot
Comment (macpijan) [reply platform-configs/include/msi-z690-common.robot:43]: https://github.com/Dasharo/open-source-firmware-validation/issues/540

File: lib/bios/menus.robot
Comment (macpijan) [reply lib/bios/menus.robot:432]: https://github.com/Dasharo/open-source-firmware-validation/issues/540

File: lib/bios/menus.robot
Comment (macpijan) [reply lib/bios/menus.robot:446]: https://github.com/Dasharo/open-source-firmware-validation/issues/540

File: lib/bios/menus.robot
Comment (JanPrusinowski) [reply lib/bios/menus.robot:416]: Fortunately resetting options to default does not turn off serial redirection until DUT is reset. So the option can be turned on before so.
My understanding was that atm only PiKVM enabled platforms use serial redirection. However it is a good idea to differentiate where this option should be enabled by some other way.

File: lib/bios/menus.robot
Comment (JanPrusinowski) [reply lib/bios/menus.robot]: I have removed the double check in: 49bc96d8b5835d5c5efc92929eeaa6e8248c23db

File: lib/bios/menus.robot
Comment (JanPrusinowski) [reply lib/bios/menus.robot]: In each test I have come across which uses this KWD there is a 'Save and Reset' KWD used afterwards. As I understand pressing 'F9' wont save settings automatically. So there is a need to save setings anyway after using this kwd.

File: dasharo-compatibility/cpu-cores-count.robot
Comment (macpijan) [reply dasharo-compatibility/cpu-cores-count.robot:61]: Info: resolved by: https://github.com/Dasharo/open-source-firmware-validation/commit/836169f6940ec10d65c81c4c3a302f8c06905068

File: dasharo-compatibility/esp-scanning.robot
Comment (macpijan) [reply dasharo-compatibility/esp-scanning.robot:30]: https://github.com/Dasharo/open-source-firmware-validation/blob/develop/lib/esp-scanning-lib.robot#L65

File: dasharo-security/rebar.robot
Comment (macpijan) [inline dasharo-security/rebar.robot:23]: Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

File: dasharo-security/tcg-opal-disk-password.robot
Comment (macpijan) [inline dasharo-security/tcg-opal-disk-password.robot:50]: @JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

File: dasharo-security/verified-boot.robot
Comment (macpijan) [reply dasharo-security/verified-boot.robot:162]: @JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

File: keywords.robot
Comment (macpijan) [inline keywords.robot:215]: 50/50 - TBD?

File: keywords.robot
Comment (macpijan) [reply keywords.robot:1732]: @JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

File: lib/options/uefi-setup-menu.robot
Comment (macpijan) [inline lib/options/uefi-setup-menu.robot:62]: @JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

File: lib/secure-boot-lib.robot
Comment (macpijan) [reply lib/secure-boot-lib.robot:151]: @JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

File: platform-configs/msi-pro-z690-a-wifi-ddr4.robot
Comment (macpijan) [inline platform-configs/msi-pro-z690-a-wifi-ddr4.robot:10]: @JanPrusinowski Likely we need this to be ported to: https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

File: dasharo-security/verified-boot.robot
Comment (krystian-hebel) [reply dasharo-security/verified-boot.robot:162]: I was hoping that the log would show per-line timestamps, but SSH Library hides that information. Let's go with 20m then.

File: lib/options/uefi-setup-menu.robot
Comment (JanPrusinowski) [reply lib/options/uefi-setup-menu.robot:133]: It is copied from `lib/options/dcu.robot`. As I understand this file is included when options are changed by DCU. When not the `uefi-setup-menu.robot` is included. This command should be mirrored in both files... This kwd might not fit here but it is a DCU issue... As you noticed - it is not the only kwd which shouldnt be there. DCU might need a refactor to make order with those KWDs.

File: lib/bios/menus.robot
Comment (JanPrusinowski) [reply lib/bios/menus.robot]: Enabling serial wont be merged with this branch. Instead it is a part of a seprate task: https://github.com/Dasharo/open-source-firmware-validation/issues/540

File: platform-configs/include/msi-z690-common.robot
Comment (JanPrusinowski) [reply platform-configs/include/msi-z690-common.robot]: This was fixed in https://github.com/Dasharo/open-source-firmware-validation/commits/regression-msi/

File: lib/bios/menus.robot
Comment (JanPrusinowski) [reply lib/bios/menus.robot:431]: Maciej created a seprate task for this issue: https://github.com/Dasharo/open-source-firmware-validation/issues/540

File: platform-configs/include/msi-z690-common.robot
Comment (JanPrusinowski) [reply platform-configs/include/msi-z690-common.robot:93]: This is a part of: https://github.com/Dasharo/open-source-firmware-validation/issues/540

File: lib/bios/menus.robot
Comment (BeataZdunczyk) [reply lib/bios/menus.robot:416]: @JanPrusinowski please link the issue here

File: lib/secure-boot-lib.robot
Comment (BeataZdunczyk) [reply lib/secure-boot-lib.robot:151]: @JanPrusinowski please link where changes were added.

File: platform-configs/msi-pro-z690-a-wifi-ddr4.robot
Comment (BeataZdunczyk) [reply platform-configs/msi-pro-z690-a-wifi-ddr4.robot:32]: @JanPrusinowski broken link ^

File: lib/options/uefi-setup-menu.robot
Comment (BeataZdunczyk) [reply lib/options/uefi-setup-menu.robot:133]: If we don't want to resolve this issue here, maybe we could create an issue for it @JanPrusinowski ?

File: lib/options/uefi-setup-menu.robot
Comment (BeataZdunczyk) [reply lib/options/uefi-setup-menu.robot:62]: @JanPrusinowski link?

File: lib/secure-boot-lib.robot
Comment (JanPrusinowski) [reply lib/secure-boot-lib.robot:151]: https://github.com/Dasharo/open-source-firmware-validation/commit/133b843580206f791fd72831bbb26e02ecfbaf75

File: platform-configs/msi-pro-z690-a-wifi-ddr4.robot
Comment (JanPrusinowski) [reply platform-configs/msi-pro-z690-a-wifi-ddr4.robot:32]: Works for me... But I edited it so that it is wraped.

File: lib/options/uefi-setup-menu.robot
Comment (JanPrusinowski) [reply lib/options/uefi-setup-menu.robot:62]: https://github.com/Dasharo/open-source-firmware-validation/commit/904dd1432f23058a795faff172a352c1ac3785a7

File: platform-configs/msi-pro-z690-a-wifi-ddr4.robot
Comment (JanPrusinowski) [reply platform-configs/msi-pro-z690-a-wifi-ddr4.robot:10]: https://github.com/Dasharo/open-source-firmware-validation/commit/c2aec8eb46036528534b53245960e8f184f8ddca

File: dasharo-security/tcg-opal-disk-password.robot
Comment (JanPrusinowski) [reply dasharo-security/tcg-opal-disk-password.robot:50]: https://github.com/Dasharo/open-source-firmware-validation/commit/edda9f798c950011a1acedf7cc46e905003570c3

File: dasharo-security/rebar.robot
Comment (JanPrusinowski) [reply dasharo-security/rebar.robot:23]: https://github.com/Dasharo/open-source-firmware-validation/commit/904dd1432f23058a795faff172a352c1ac3785a7

File: lib/options/uefi-setup-menu.robot
Comment (JanPrusinowski) [reply lib/options/uefi-setup-menu.robot:133]: Issue: https://github.com/Dasharo/open-source-firmware-validation/issues/547

**Comment by BeataZdunczyk** [comment]
Lots of threads to resolve. Could you please @macpijan , and @krystian-hebel recheck your threads so we can resolve them and merge this PR?

**Comment by JanPrusinowski** [comment]
Everything needed from this PR was moved to: https://github.com/Dasharo/open-source-firmware-validation/pull/550

**Pull request closed without merging**
